pandas/tests/util/test_assert_frame_equal.py

```diff
@@ -285,3 +287,15 @@ def test_allows_duplicate_labels():
 
     with pytest.raises(AssertionError, match="<Flags"):
         tm.assert_frame_equal(left, right)
+
+
+def test_assert_frame_equal_checks_index_exactly_twice():
```

Comment:
i dont think this necessarily needs a test, _maybe_ an asv.  as long as there is a comment in tm.assert_frame_equal about why we are doing what we're doing that should be enough

Reply:
I am removing the test because it relies on ``unittest.mock``, which is not accepted by the CI check.